[Only Test] Add Symbiosis Swaps (#506)
rsksmart/devportal: rif/identity/data-vault/architecture/client.md

##### Auth Manager

> It manages authentication according to the [DID Auth protocol](../../../rlogin/libraries/express-did-auth). It is in charge of emitting the necessary events to be signed by the user and store the generated tokens. Will associate the generated tokens to the current DID so it allows multiple sessions with different DIDs.

Reviewer: ```suggestion
It manages authentication according to the [DID Auth protocol](../../../rlogin/libraries/express-did-auth). It is in charge of emitting the necessary events to be signed by the user and store the generated tokens. It will associate the generated tokens to the current DID so it allows multiple sessions with different DIDs.
```

It is instantiated with a `DIDAuthConfig` object, which contains the following fields:

> - `serviceUrl: string`: the IPFS Centralized Data Vault Service url, the user will be logged in to that url

Reviewer: ```suggestion
- `serviceUrl: string`: the IPFS Centralized Data Vault Service URL, which enables the user to be logged in.
```

##### Encryption Manager

It is in charge of managing the encryption of the saved information and the decryption of the received information.

> It also decides whether to decrpyt or not the content based on the current status, it will check if is encrypted before trying to decrypt it.

Reviewer: ```suggestion
It also decides whether or not to decrypt the content based on the current status, it also checks if it's encrypted before trying to decrypt it.
```

It receives an `EncryptionManagerConfig` with the following fields:

> - `getEncryptionPublicKey?: () => Promise<string>`: the method used to get the user encryption public key. That public key will be used to encrypt the content before sending it to the service to be stored. [Metamask](https://docs.metamask.io/guide/rpc-api.html#eth-getencryptionpublickey) example: `() => window.ethereum.request.request({ method: 'eth_getEncryptionPublicKey', params: [address] })`. If it is not provided, the content will not be encrypted prior to save it in IPFS.

Reviewer: ```suggestion
- `getEncryptionPublicKey?: () => Promise<string>`: the method used to get the user encryption public key. That public key will be used to encrypt the content before sending it to the service to be stored. [Metamask](https://docs.metamask.io/guide/rpc-api.html#eth-getencryptionpublickey) example: `() => window.ethereum.request.request({ method: 'eth_getEncryptionPublicKey', params: [address] })`. If it is not provided, the content will not be encrypted prior to saving it in IPFS.
```

> - `decrypt: (hexCypher: string) => Promise<string>`: the method used to decrypt data stored in the service. If using [Metamask](https://docs.metamask.io/guide/rpc-api.html#eth-decrypt) example: `(hexCypher: string) => window.ethereum.request({ method: 'eth_decrypt', params: [hexCypher, address] })`, then this method will prompt the user to decrypt content in the wallet. If no Metamask wallet present, we strongly recommend to create a function that alert the user that the content will not be decrypted because the wallet does not include that feature.

Reviewer: ```suggestion
- `decrypt: (hexCypher: string) => Promise<string>`: the method used to decrypt data stored in the service. If using [Metamask](https://docs.metamask.io/guide/rpc-api.html#eth-decrypt) example: `(hexCypher: string) => window.ethereum.request({ method: 'eth_decrypt', params: [hexCypher, address] })`, then this method will prompt the user to decrypt content in the wallet. If there's no Metamask wallet present, we strongly recommend creating a function that alerts the user that the content will not be decrypted because the wallet does not include that feature.
```
